### ISSUE TITLE
[Feature] CUDA Python 1.0

### ISSUE THREAD

**Issue by zeydsama** [Haoming02/sd-webui-forge-classic#1436]
https://developer.nvidia.com/blog/cuda-python-1-0-stable-apis-one-foundation-full-platform-access/
reff: https://github.com/Haoming02/sd-webui-forge-classic/discussions/1432

**Comment by lz-55** [comment]
Can reduce generation time by 2-2.5 seconds from my tests, that's not bad but will probably be a much bigger benefit with training.

**Comment by Haoming02** [comment]
> Can reduce generation time by 2-2.5 seconds from my tests

What modification did you do?

**Comment by lz-55** [comment]
ChatGPT describes it as more of a "CUDA execution accelerator" and i think it can probably be taken much further (and I'm currently looking into that), but feel free to have a look:

[sd-forge-cuda-engine-v0.5.9-alpha24-forge-neo-.zip](https://github.com/user-attachments/files/31625797/sd-forge-cuda-engine-v0.5.9-alpha24-forge-neo-.zip)

**Comment by Haoming02** [comment]
OK, that's way more expansive than I'd expected

**Comment by Haoming02** [comment]
Though, I get no speed up on my end

**Comment by lz-55** [comment]
Generating at 1216x2112 for 12 steps there is consistently close to 2 seconds difference for me, goes from around 46 seconds with it off and 44 when i enable it, do a generation so it caches and then the next gen is lower than with it off.

**Comment by Haoming02** [comment]
Which model and which quant?

I tried Krea2 (Turbo) `int8`, and both finish in 5 seconds

**Comment by lz-55** [comment]
Krea 2 Turbo INT8, but now im trying things at 1024 with 8-12 steps it doesn't seem to make any difference, I think the reason im getting a speedup is because I have a RES4LYF extension and another that lets me chain samplers and i think this is improving the handoff between the main generation and the second sampler in those particular setups.

**Comment by zeydsama** [comment]
same.. seems no difference 

krea2:
<img width="803" height="426" alt="Image" src="https://github.com/user-attachments/assets/6b19fc85-7abb-476c-9bef-6e0e6f2cb750" />
> Steps: 8, Sampler: ER SDE, Schedule type: Beta57 (RES4LYF), CFG scale: 1, Seed: 80897312, Size: 1008x1344, Model: krea2Turbo_v10-int4_convrot, Model hash: 6afa63dcad, Module 1: Qwen2D_VAE, Module 2: qwen3-vl-4b-heretic_int8, RNG: GPU, Lora hashes: "Krea2_TextFusion_Blocks_Only_v3_000007900: 29216198500c", Krea2 NegPiP value strength: 1.0, Krea2 NegPiP refiners: True, Emphasis: No norm, spec_w: 0.8, spec_m: 2, spec_lam: 0.1, spec_window_size: 2, spec_flex_window: 0, spec_warmup_steps: 5, spec_stop_caching_step: 0.95, Krea2 Conditioning Preset: subtle, Krea2 Conditioning Weights: "1, 1, 1, 1, 1, 1, 1, 1.5, 2, 1, 1.5, 1", Krea2 Conditioning Multiplier: 1.0, Krea2 Conditioning Renormalize: True, torch compile: dynamic, INT8 Fused GEMM: true, CUDA Engine: 0.5.9-alpha24, CUDA Engine Autotune: True, CUDA Engine Validate: True, CUDA Engine min win: 10.0%, CUDA Engine Forge GEGLU: True, CUDA Engine Diffusers gated: True, CUDA Engine GroupNorm SiLU: True, CUDA Engine Transformer residual norm: True, CUDA Engine ResBlock scale shift: True, CUDA Engine Attention autotune: True, CUDA Engine Krea native: True, CUDA Engine Krea RMSNorm: True, CUDA Engine Krea SwiGLU: True, CUDA Engine Krea block fusion: True, CUDA Engine Krea INT8: True, CUDA Engine cuBLASLt GEMM: False

anima:
<img width="797" height="379" alt="Image" src="https://github.com/user-attachments/assets/c97dcdcc-53aa-4d96-b42b-f2231fd65882" />
> Steps: 35, Sampler: ER SDE, Schedule type: Beta57 (RES4LYF), CFG scale: 5, Shift: 3, Seed: 1221884553, Size: 896x1120, Model: ragAnimaMix_v02_noShinySlopAiSkin-int8_convrot, Model hash: 849f0cbad2, Module 1: Qwen2D_VAE-bf16, Module 2: Huihui-Qwen3-0.6B-abliterated-v2, RNG: CPU, Lora hashes: "highres_fix: 9c051d36648f", Emphasis: No norm, spec_w: 0.05, spec_m: 8, spec_lam: 0.6, spec_window_size: 2, spec_flex_window: 0.1, spec_warmup_steps: 12, spec_stop_caching_step: 0.95, torch compile: max-autotune-no-cudagraph, INT8 Fused GEMM: true, CUDA Engine: 0.5.9-alpha24, CUDA Engine Autotune: True, CUDA Engine Validate: True, CUDA Engine min win: 10.0%, CUDA Engine Forge GEGLU: True, CUDA Engine Diffusers gated: True, CUDA Engine GroupNorm SiLU: True, CUDA Engine Transformer residual norm: True, CUDA Engine ResBlock scale shift: True, CUDA Engine Attention autotune: True, CUDA Engine Krea native: True, CUDA Engine Krea RMSNorm: True, CUDA Engine Krea SwiGLU: True, CUDA Engine Krea block fusion: True, CUDA Engine Krea INT8: True, CUDA Engine cuBLASLt GEMM: False

**Comment by Haoming02** [comment]
From my limited understanding, the `CCCL` package thingy is basically at the same level of `PyTorch` in terms of tech stack. So you will need to basically rewrite the entire backend to use it instead.

> **i.e.** they are both "wheels" for a car, you cannot use both at the same time

**Comment by Richardpanacek** [comment]
Is there any difference is **VRAM/RAM usage**? 

Maybe your PCs are too good for seeing the benefit.

**Comment by lz-55** [comment]
I've pretty much 100% confirmed now that this is only speeding up the handoff between the first and second sampler in my setup, that's why it seems to do nothing.

<img width="1920" height="1622" alt="Image" src="https://github.com/user-attachments/assets/a51cad0a-d018-4e7f-a915-7cd6f083a046" />

**Comment by Haoming02** [comment]
> you will need to basically rewrite the entire backend to use it instead

Then yeah, I probably won't do anything with it. Unless ComfyUI adapts it somehow.
